### PR TITLE
Graph optimize

### PR DESCRIPTION
**Pull request by JulianWielga**
fix: minor bugs
feat: highlight link to insert
feat: layout only selected
feat: delete group with nodes
feat: copy/paste groupped nodes
fix: layout and render optimize
fix: panzoom graph plugin minor bugs
feat: layout pasted
fix: fit and zoom after layout minor bugs
refactor: some more typescript
refactor: selectors and action creators

### CODE REVIEW

File: ui/client/components/graph/GraphPartialsInTS/drawGraph.ts [hunk 1]
```diff
@@ -24,9 +23,9 @@ export function drawGraph(
   performance.mark("redrawing start")
   this.redrawing = true
 
-  const nodesWithGroups = NodeUtils.nodesFromProcess(process, expandedGroups)
-  const edgesWithGroups = NodeUtils.edgesFromProcess(process, expandedGroups)
-  const groups = NodeUtils.getExpandedGroups(process, expandedGroups)
+  const nodesWithGroups = NodeUtils.nodesFromProcess(process)
```
Comment (lciolecki) [inline ui/client/components/graph/GraphPartialsInTS/drawGraph.ts:26]: This method should be named `nodesWithGroupsFromProcess` and below method also should be chaged.

File: ui/client/components/graph/GraphPartialsInTS/drawGraph.ts [hunk 2]
```diff
@@ -36,15 +35,17 @@ export function drawGraph(
...
-  const newCells = cells.filter(cell => !graph.getCell(cell.id))
-  const deletedCells = graph.getCells().filter(oldCell => !cells.find(cell => cell.id === oldCell.id))
+  const currentCells = graph.getCells()
+  const currentIds = currentCells.map(c => c.id)
+  const newCells = cells.filter(cell => !currentIds.includes(cell.id))
+  const deletedCells = currentCells.filter(oldCell => !cells.find(cell => cell.id === oldCell.id))
   const changedCells = cells.filter(cell => {
     const old = graph.getCell(cell.id)
-    return !isEqual(old?.get("definitionToCompare"), cell.get("definitionToCompare"))
+    return old && !isEqual(old.get("definitionToCompare"), cell.get("definitionToCompare"))
```
Comment (lciolecki) [inline ui/client/components/graph/GraphPartialsInTS/drawGraph.ts:44]: You missed `old?.get`  I mean `?`

File: ui/client/components/graph/PanZoomPlugin.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import {dia} from "jointjs"
+import {debounce} from "lodash"
 import svgPanZoom from "svg-pan-zoom"
 import {CursorMask} from "./CursorMask"
 import {Events} from "./joint-events"
 
-type EventData = { panStart?: { x: number, y: number } }
+type EventData = { panStart?: { x: number, y: number, touched?: boolean } }
```
Comment (lciolecki) [inline ui/client/components/graph/PanZoomPlugin.ts:7]: What does mean touched? Maybe better will be `chaged`?

File: ui/client/reducers/graph/utils.ts [hunk 4]
```diff
@@ -151,7 +151,7 @@ export function addNodesWithLayout(state: GraphState, {nodes, layout}: ReturnTyp
 export function removeSubprocessVersionForLastSubprocess(processToDisplay: Process, idToDelete: NodeId) {
   const subprocessVersions = processToDisplay.properties.subprocessVersions
   const nodeToDelete = processToDisplay.nodes.find(n => n.id === idToDelete)
-  if (nodeToDelete.type === "SubprocessInput") {
+  if (nodeToDelete?.type === "SubprocessInput") {
```
Comment (lciolecki) [inline ui/client/reducers/graph/utils.ts:154]: Can we have node without type? And I think node's type should be enum ;)

**Review by lciolecki — APPROVED** [review]
Looks good, but please look at my comments.

File: ui/client/reducers/graph/utils.ts (hunk 4 above)
Comment (JulianWielga) [reply ui/client/reducers/graph/utils.ts:154]: find may return nothing.
no it shouldn't be enum, brings nothing and possibly infinite list

File: ui/client/components/graph/PanZoomPlugin.ts (hunk 3 above)
Comment (JulianWielga) [reply ui/client/components/graph/PanZoomPlugin.ts:7]: touched - dirty - used - I need this state to filter out random movements

File: ui/client/components/graph/GraphPartialsInTS/drawGraph.ts (hunk 1 above)
Comment (JulianWielga) [reply ui/client/components/graph/GraphPartialsInTS/drawGraph.ts:26]: why, what we could get from that change?

**Pull request merged**